pkg/staticnetworkconfig/generator.go

```diff
@@ -106,14 +107,10 @@ func (s *StaticNetworkConfigGenerator) formatNMConnection(nmConnection string) (
 		return "", fmt.Errorf("%s", msg)
 	}
 
-	mac := ifaceName.String()
-	splitMac := []string{}
-	i := 0
-	for i < len(mac) {
-		splitMac = append(splitMac, mac[i:i+2])
-		i += 2
+	mac, err := s.validateAndCalculateMAC(ifaceName.String())
```

Comment:
will it be relevant for all of the network configurations (including vlan names) ?

Reply:
partly. i am currently opening a ticket for vlan handling, this function will be used there too